### PR TITLE
[fix] fix problem with main require compatibility

### PR DESCRIPTION
**Pull request by kele6ra**
Reverse basic version of index file. Add additional file for custom binding.

### CODE REVIEW

**Review by rzr — COMMENTED** [review]
more review welcome

**Comment by kele6ra** [comment]
@Apollon77 welcome to review =)

File: README.md [hunk 1]
```diff
@@ -42,6 +42,12 @@ noble.on('discover', async (peripheral) => {
   await peripheral.disconnectAsync();
   process.exit(0);
 });
+```
+## Use Noble With BLE5 Extended Features With HCI 
+
+```javascript
+const noble = require('@abandonware/noble/with-custom-binding')({extended: false});
```
Comment (Apollon77) [inline README.md:49]: should exteded be not true in this example?

**Review by Apollon77 — COMMENTED** [review]
Rest LGTM

File: README.md (hunk 1 above)
Comment (kele6ra) [reply README.md:49]: Only if we don't want to use ble5 features =) I will change it.

Comment (Apollon77) [reply README.md]: ;-) I think the example exactly "want" that ;-)) Thank you

**Review by Apollon77 — APPROVED** [review]

**Comment by Apollon77** [comment]
@bwp91 Looks also good to you? @AlCalzone too if you find time

**Comment by Apollon77** [comment]
@AlCalzone approved too ... let's merge it. I think best we can do without breaking.

File: index.js [hunk 2]
```diff
@@ -1,6 +1,4 @@
-module.exports = function (options) {
-  const Noble = require('./lib/noble');
-  const bindings = require('./lib/resolve-bindings')(options);
+const Noble = require('./lib/noble');
+const bindings = require('./lib/resolve-bindings');
```
Comment (xqin) [inline index.js:2]: this should be `const bindings = require('./lib/resolve-bindings')();`

Comment (xqin) [reply index.js:2]: Or modify the code of the entire file to:

```
module.exports = require('./with-custom-binding')();
```

**Comment by xqin** [comment]
And can we unpublish `1.9.2-17` and `1.9.2-18` version  from npmjs registry first?

Since it's a breaking change, shouldn't publish as `1.9.2-*` serial.

File: index.js (hunk 2 above)
Comment (kele6ra) [reply index.js]: Done

**Comment by Apollon77** [comment]
LGTM (still) :-)

@xqin: I assume it is too late for an unpublish, so we should focus on publish of the fixed version

**Comment by xqin** [comment]
@Apollon77  OK :)

**Comment by bwp91** [comment]
@Apollon77 can confirm with this PR I no longer have errors with the `.on()` issues.
So happy for this to be merged in.

As a side note, I do seem to get this in the log,
```
ERROR:  Could not load manufacturer list from "./manufactures-hex.json".
        Please see "/home/pi/govee/node_modules/@abandonware/noble/scripts/manufactures/README.md" for instructions.
ERROR:  Could not load manufacturer list from "./manufactures-dec.json".
        Please see "/home/pi/govee/node_modules/@abandonware/noble/scripts/manufactures/README.md" for instructions.
```

**Comment by Apollon77** [comment]
@rzr Then we should merge and release that please!

For that error, I digged a bit in code ... in fact introduced in https://github.com/abandonware/noble/pull/154/files ... so it adds the "option" to use such files and loads them from the noble lib (makes that sense? because someone would need to monkey patch these files in ... @pixel-shock how this is thought?) And then logs console.error if files are not found which is veeery misleading for an (as it seems) optional feature. 

The script to add the files is added in unmerged #157 and #158 (open) and #156 arre also somehow connected here.

But #157 adds that as dev dependency ... but it misses somehow to include this into the release/build process so even if these would be merged the logged error would be the same

And yes because it seems that the release of -16 in september did not worked it went to live now.

@rzr @pixel-shock Please revert that one commit or finish the topic. Thank you!

**Comment by Apollon77** [comment]
@rzr
with all these issues what do you think about setting "latest" dist-tag of noble to the last version ... this would be "kind of an unpublish" and should prevent too many users upgrade until we sorted all that out.

Command: `npm dist-tag add @abandonware/noble@1.9.2-15 latest`

**Comment by rzr** [comment]
ok I can try this

**Comment by Apollon77** [comment]
> btw: Is this change reviewed and tested ?

When I understood @kele6ra and @bwp91 correctly then yes

**Comment by kele6ra** [comment]
> > btw: Is this change reviewed and tested ?
> 
> When I understood @kele6ra and @bwp91 correctly then yes

Yes, we made ble5 extended features support for our project. 
It have been tested with rpi3 and zephyr hci dongle (Raspberry Pi OS last version) and in win10, win11, Ubuntu 20.04 int the host PC.

**Comment by Apollon77** [comment]
@rzr how should we continue here to get that merged??

**Comment by Apollon77** [comment]
@rzr thank you for merging. would you trigger a release or do you need a PR for that?

**Comment by Apollon77** [comment]
Hm ... no not really because the whole topic around https://github.com/abandonware/noble/pull/283#issuecomment-1362530533 also needs to be adressed

**Comment by Apollon77** [comment]
I got more bad news. Several users report increased cpu usage (climbing over time) with the new version.

**Comment by kele6ra** [comment]
> I got more bad news. Several users report increased cpu usage (climbing over time) with the new version.

In my feature I haven't changed the basic one.

**Comment by Apollon77** [comment]
No idea for now ... will try to go to all commits merged since last version and will now rollback/pinn all my projects to the -15 :-(

**Comment by Apollon77** [comment]
I had a scan through all commits since -15 and did not saw anything obvious ... (but also did not checked the windows stuff in depth because most reports are linux that I got so far :-(
I fear it will be a big fun to find the reason using git bisect unless we find a better "proof" to see the issue earlier :-(

**Comment by Apollon77** [comment]
@kele6ra with your code knowledge ... are there any places where we iterate over a list of e.g. "discovered devices" or addresses or anything like that? Or an array find in such? maybe something bacse more expensive what is don in the iterations or find or such ... or we have lists that we splice/slice or such

I Ask this because the reports on performance issues are that they show up after some time and it "slowly increases" ... SO my best guess is something like this that a list gets big over time (e.g. from discovered devices if someone lives near a street and does so collect many adresses. In fact it is NOT a memory issue, so it is not in fact about the number of found stuff, but that something "costly" is done with it

**Comment by Apollon77** [comment]
My feeling goes for checks in the direction of scanning and discovering - because the reported usages are 9x% aboout that and not that much about "intercations"

**Comment by Apollon77** [comment]
@kele6ra Question: Why you lowered the le connection timeout?

https://github.com/abandonware/noble/commit/b67eea246f719947fc45b1b52b856e61637a8a8e#diff-ea21f5c8b475a4af7e34ea3a01714af0731d83b5706497d089185b00b0fe0642L351-R485

c8->2a (thats lowered a looootttt)

**Comment by kele6ra** [comment]
> @kele6ra Question: Why you lowered the le connection timeout?
> 
> [b67eea2#diff-ea21f5c8b475a4af7e34ea3a01714af0731d83b5706497d089185b00b0fe0642L351-R485](https://github.com/abandonware/noble/commit/b67eea246f719947fc45b1b52b856e61637a8a8e#diff-ea21f5c8b475a4af7e34ea3a01714af0731d83b5706497d089185b00b0fe0642L351-R485)
> 
> c8->2a (thats lowered a looootttt)

Hello, I set parameters from the bluez stack. 
I have started operations with bluez and looked with btmon. Then I implement that commands in the noble. 
That's why in some places I have just copied parameters. In some places (if I couldn't find the documentation) I have checked different values before I got stable work.

**Comment by kele6ra** [comment]
I haven't made any modifications and worked with code from the August. But I will check it again in a few days.

**Comment by rzr** [comment]
to release just commit a change in same form of previous one

**Comment by Apollon77** [comment]
I think I found the reason for the high CPU usage ... https://github.com/abandonware/node-bluetooth-hci-socket/commit/00bc1578193d7d3045537e0fd76cda983cf1390d#r94830612

Fixing the multiple-calls is relatively easy ... but how to handle the case that "stop" is never called on the socket

**Comment by rzr** [comment]
timeout ?

**Comment by Apollon77** [comment]
In fact the interval is calling some cleanup logic in the socket instance ... In theory it would be better to have that interval onkly triggered if there actually is a con ectionm so cleanup could do something ... but thats in C code ...

Maybe it is alsl not that much of a deal if we prevent just the muti-intervals.

When I understand the code then Noble(hci socket)Bindings have ONE hci instance which has ONE hci socket ... so it this is really that way then without the duplicate intervals we only have exactly ONE interval running for socket cleanups. If noone else understands the code differently I will provide a PR ... --> https://github.com/abandonware/node-bluetooth-hci-socket/pull/45

**Comment by Apollon77** [comment]
Ok (and yes this went off topic a lot and sorry for that but lets continue in one place):

Under the assumption that we have the performance issue tackled with the PR ... what do we do with https://github.com/abandonware/noble/pull/283#issuecomment-1362530533 ??

@pixel-shock @rzr Revert or complete? I have no the time to find oput how to best complete that one.
We can:
* Revert it
* leave in but remove the console error outputs when files are not existing and diocument it as "optional feature to use" (how ever that ever should end up on a end user syytem then in the right place ......)
* finalize it means adjusting the build process and find out hopw to get that officially and such

I personally would vote for revert of #154 and closing of the two open PRs and @pixel-shock should bring it in again after discussions on how to utilize this. @rzr your decision as the maintainer ... If anyone else have opinions, speak up please.

We need to get that mess sorted out to get a proper release as future basis

**Pull request merged**